File: app/src/main/java/com/lazycouple/restapiclient/ui/base/BaseFragment.java
package com.lazycouple.restapiclient.ui.base;

import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.view.View;

import com.lazycouple.restapiclient.R;

/**
 * Created by amuyu on 2017. 6. 21..
 */

public class BaseFragment extends Fragment {


    private TabLayout tabLayout;

    public void initTabLayout(boolean visible) {
        tabLayout = (TabLayout)getActivity().findViewById(R.id.tabLayout);
        if(useTabLayout()) {
            setTabVisible(visible);
            tabLayout.removeAllTabs();
            tabLayout.clearOnTabSelectedListeners();
        }
    }


    public void setTabVisible(boolean visible) {
        tabLayout.setVisibility(visible?View.VISIBLE:View.GONE);
    }

    public TabLayout getTabLayout() {
        return tabLayout;
    }

    public boolean useTabLayout() {
        return (tabLayout!=null);
    }
}

File: app/src/main/kotlin/com/lazycouple/restapiclient/ui/base/BaseFragment.kt
package com.lazycouple.restapiclient.ui.base

import android.arch.lifecycle.LifecycleFragment
import android.support.design.widget.TabLayout
import android.view.View

import com.lazycouple.restapiclient.R

/**
 * Created by amuyu on 2017. 6. 21..
 */

open class BaseFragment : LifecycleFragment() {


    var tabLayout: TabLayout? = null
        private set

    fun initTabLayout(visible: Boolean) {
        tabLayout = activity.findViewById(R.id.tabLayout) as TabLayout
        if (useTabLayout()) {
            setTabVisible(visible)
            tabLayout!!.removeAllTabs()
            tabLayout!!.clearOnTabSelectedListeners()
        }
    }


    fun setTabVisible(visible: Boolean) {
        tabLayout!!.visibility = if (visible) View.VISIBLE else View.GONE
    }

    fun useTabLayout(): Boolean {
        return tabLayout != null
    }
}
